src/twister2/builder/build_manager.py

```diff
@@ -96,6 +105,9 @@ def build(self) -> None:
         """
         Build source code.
         """
+        if self.build_filter_processor:
```

Comment:
I think that we can move this `build_filter_processor` to `_build` method to avoid running CMake package helper in each time, when several pytest tests indicate to the same source application.